intake/source/npy.py

```diff
@@ -0,0 +1,108 @@
+from .base import DataSource, Schema
+
+
+class NPySource(DataSource):
+    container = 'ndarray'
+    name = 'numpy'
+    version = '0.0.1'
+    partition_access = True
+
+    def __init__(self, path, shape=None, chunks=None, metadata=None):
+        self.path = path
+        self.shape = shape
+        self._chunks = chunks if chunks is not None else -1
+        self.chunks = None
+        self.dtype = None
+        self._arrs = None
+        self._arr = None
+        super(NPySource, self).__init__(metadata=metadata)
+
+    def _get_schema(self):
+        from dask.bytes import open_files
+        import dask.array as da
+        if self._arr is None:
+            files = open_files(self.path, 'rb', compression=None)
+            if self.shape is None:
+                arr = NumpyAccess(files[0])
+                self.shape = arr.shape
+                self.dtype = arr.dtype
+                arrs = [arr] + [NumpyAccess(f, self.shape, self.dtype)
+                                for f in files[1:]]
+            else:
+                arrs = [NumpyAccess(f, self.shape, self.dtype)
+                        for f in files]
+            self.chunks = (self._chunks, ) + (-1, ) * (len(self.shape) - 1)
```

Comment:
Curious, why not set this in the `__init__` method and get rid of `self._chunks`?  Do we need the distinction between `self.chunks` and `self._chunks`?  Perhaps this is a convention I'm unaware of.

Reply:
I wanted to store the original input explicitly - this seems like a good idea for hygiene and serialization in general.
`self.chunks` is a computed value, and depends on the array shape, which may have been provided, but also may have come from the first of the files. Intake should perform as little work on DataSource instantiation as possible.